src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs

```diff
@@ -96,5 +124,44 @@ public static ILoggingBuilder AddOpenTelemetryLogs(this ILoggingBuilder builder)
 
         return builder;
     }
+
+    private static bool IsNet6()
+    {
+        var frameworkDescription = FrameworkDescription.Instance;
+        return frameworkDescription.Name == ".NET" && frameworkDescription.ProductVersion.StartsWith("6");
+    }
+
+    private static bool IsHostServiceCollection(IServiceCollection builderServices)
+    {
+        // check if assembly is loaded before trying to get type,
+        // in order to avoid triggering an unnecessary load
+        var assemblies = AppDomain.CurrentDomain.GetAssemblies();
```

Comment:
We can avoid this check as Type.GetType won't load the assembly if it's already loaded.

Reply:
removed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3003/commits/0c129a81f0e8f710bfbfe45f1bcab2c30543d7b9